### PR TITLE
[WIN32K:ENG] EngInitializeSafeSemaphore(): Do not busy wait

### PR DESCRIPTION
**Pull request by SergeGautherie**
Minor improvement.

### CODE REVIEW

**Comment by learn-more** [comment]
Why?
What does this fix?

**Review by katahiromz — COMMENTED** [review]
Initialization timing of `Interval` can be moved.

**Comment by SergeGautherie** [comment]
> Why? What does this fix?

Why not? No fix, minor improvement.

> can be moved

Right. Moved.

**Review by katahiromz — COMMENTED** [review]
Okay.

**Review by learn-more — CHANGES_REQUESTED** [review]
> > Why? What does this fix?
> 
> Why not? No fix, minor improvement.
> 

That is subjective.
Without a valid reason of changing stuff like this, it makes no sense to change random stuff to your own liking.

**Comment by SergeGautherie** [comment]
> subjective

Well, I don't understand what is subjective about busy wait being unwanted...
Similar (goal) to
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=commit&s=busy+wait
for example.

File: win32ss/gdi/eng/semaphor.c [hunk 1]
```diff
@@ -164,10 +164,16 @@ EngInitializeSafeSemaphore(
...
+        LARGE_INTEGER Interval = {0LL};
+
         /* Wait for the other thread to create the semaphore */
         ASSERT(Semaphore->lCount > 1);
         ASSERT_IRQL_LESS_OR_EQUAL(PASSIVE_LEVEL);
-        while (Semaphore->hsem == NULL);
+        while (Semaphore->hsem == NULL)
+        {
+            // Relinquish time slice.
+            KeDelayExecutionThread(KernelMode, FALSE, &Interval);
```
Comment (tkreuzer) [inline win32ss/gdi/eng/semaphor.c:175]: - [ ] Use ZwYieldExecution()

**Review by tkreuzer — CHANGES_REQUESTED** [review]

File: win32ss/gdi/eng/semaphor.c (hunk 1 above)
Comment (SergeGautherie) [reply win32ss/gdi/eng/semaphor.c]: Thanks for the reminder. Done.

**Review by tkreuzer — APPROVED** [review]

**Comment by GeoB99** [comment]
You should provide some explanation on what it has exactly improved since you're the author of this PR and henceforth of your code. We cannot simply take a guess or playing mind games on trying to understand your intention of this patch,

When you come back with a proper description of your patch, I will open the PR back.

**Pull request closed without merging**